elasticsearch/src/main/scala/com/snowplowanalytics/snowplow/loader/clients/ElasticsearchSenderHTTP.scala

```diff
@@ -91,21 +94,22 @@ class ElasticsearchSenderHTTP(
 
     val newFailures: List[EmitterInput] = if (actions.nonEmpty) {
       futureToTask(client.execute(bulk(actions)))
-        // we rety with linear backoff if an exception happened
+        // we retry with linear back-off if an exception happened
         .retry(delays, exPredicate(connectionAttemptStartTime))
-        .map { bulkResponse =>
-          bulkResponse.items.zip(records)
+        .map {
+          case Right(bulkResponseResponse: RequestSuccess[BulkResponse]) => bulkResponseResponse.result.items.zip(records)
```

Comment:
you can do `case Right(bulkResponse)` and `case Left(requestFailure)` no need to specify the types